libs/commons/Logs_helpers.ml

```diff
@@ -110,6 +114,41 @@ let setup_logging ?(skip_libs = default_skip_libs) ~force_color ~level () =
          | "application" -> ()
          | s -> failwith ("Logs library not handled: " ^ s))
 
+(*****************************************************************************)
+(* Test helpers *)
+(*****************************************************************************)
+
+(* TODO: use save_excursion on disable_set_reporter and Logs.reporter
+ * to restore them
+ *)
+let with_mocked_logs ~f ~final =
+  Logs.set_reporter
+    {
+      Logs.report =
+        (fun (_src : Logs.src) (_level : Logs.level) ~over k msgf ->
+          let k _ =
+            over ();
+            k ()
+          in
+          msgf (fun ?header:_ ?tags:_ fmt ->
+              let (ppf : Format.formatter) = Format.str_formatter in
+              Format.kfprintf k ppf fmt));
+    };
+  (* TODO? use Fun.finalize? or save_excursion *)
+  disable_set_reporter := true;
+  (* f() might call setup_logging() internally, but this will not
+   * call Logs.set_reporter and override the reporter we set above
+   * thx to disable_set_reporter
+   *)
+  let res = f () in
+  disable_set_reporter := false;
+  let content = Format.flush_str_formatter () in
+  final content res
```

Comment:
another strategy could be to use a pipe -- and follow how mirage-logs does its testing (see https://github.com/mirage/mirage-logs/blob/main/test/test.ml). but imho the approach taken here (logging to a buffer) is less error-prone (esp. since there is as well easy_logging around which may output something to stdout/stderr)